packages/autotask-client/src/bin.ts

```diff
@@ -1,41 +1,159 @@
+    const client = initClient();
+    console.error(`Uploading code for autotask ${autotaskId} from ${path}...`);
+    await client.updateCodeFromFolder(autotaskId, path);
+  } catch (error) {
+    const err = error as Error | AxiosError;
+    console.error(`Error updating Autotask code: ${err.message}`);
+    process.exit(1);
+  }
 }
 
-async function main() {
-  if (argv[2] !== 'update-code' || !argv[3] || !argv[4]) {
-    printUsage();
-    return;
+/**
+ * Utilizes the Autotask API to poll for new runs and print them out.
+ */
+async function tailRuns() {
+  const autotaskId = argv[3];
+  const logSavePath = argv[4];
+  let logSaveDest: string | undefined;
+  let stream: WriteStream | undefined;
+  if (logSavePath) {
+    validatePath(logSavePath);
+    logSaveDest = `${logSavePath}/${autotaskId}-runs.log`;
+    stream = createWriteStream(logSaveDest, { flags: 'a' });
   }
 
   try {
-    await updateCode();
-  } catch (err) {
-    console.error(`Error updating Autotask code: ${err.message}`);
+    validateId(autotaskId);
+    const client = initClient();
+    output(`\nPolling latest runs of autotask '${autotaskId}'...\n`, stream);
+    // Poll autotask runs every 2 seconds and if there are new runs, get run details and print them out.
+    let lastRun: AutotaskRunListItemResponse | undefined;
+    while (true) {
+      const newRuns = await client.listAutotaskRuns(autotaskId);
+      // If cached last run id has changed
+      if (newRuns.items[0]?.autotaskRunId !== lastRun?.autotaskRunId) {
+        lastRun = newRuns.items[0]; // cache new last run to avoid duplicates.
+
+        if (lastRun.status !== 'success') {
+          output(`\nLatest run '${lastRun.autotaskRunId}' ${lastRun.status}...`, stream);
+          lastRun = undefined; // clean up so we can check it again next time.
+        } else {
+          output(`\nLatest run '${lastRun.autotaskRunId}' ${lastRun.status}...`, stream);
+          const runDetails = await client.getAutotaskRun(lastRun.autotaskRunId);
+          // Have to make this check to satisfy Typescript.
+          if (runDetails.status === 'success') {
+            output(`\n${runDetails.decodedLogs}`, stream);
+            output(`\n------------------------------------------------------------`, stream);
+            output(`\nPolling latest runs of autotask '${autotaskId}'...\n`, stream);
+          }
+        }
+      }
+
+      await new Promise((resolve) => setTimeout(resolve, 2000));
+    }
+  } catch (error) {
+    const err = error as Error | AxiosError;
+    output(`Error on listening to Autotask runs: ${err.message}`, stream);
     process.exit(1);
   }
 }
 
+/**
+ * Utilizes the Autotask API to trigger autotask run manually.
+ */
+async function executeRun() {
+  const autotaskId = argv[3];
+  try {
+    validateId(autotaskId);
+    const client = initClient();
+    console.log(`Executing autotask run for autotask '${autotaskId}'...`);
+    // TODO: pass in params maybe? Do we have any case for that?
```

Comment:
Is this `TODO` deprecated or is it suggesting something like reading input from the CLI? I think the existing implementation seems to work fine.

Reply:
yes, I put it there to bring attention to that I'm not sure if we should accept the custom data param from the CLI or not.

The API does accept arbitrary data, but I'm not aware of any situation when this is handled by the cloud functions or any use case for it.